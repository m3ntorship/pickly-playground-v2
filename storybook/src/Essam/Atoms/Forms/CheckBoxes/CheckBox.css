/* Checkboxes */
.checkbox {
  @apply relative;
}

.checkbox .checkbox-input {
  @apply absolute top-0 left-0 opacity-0;
  z-index: -1;
}

.checkbox .checkbox-input:checked + .check-box .check-mark {
  @apply block;
}

.checkbox .checkbox-input:checked + .check-box {
  @apply bg-primary hover:bg-primary-hover;
}

.checkbox .checkbox-input:focus + .check-box {
  @apply border-grey-shd3 border-2;
}

.checkbox .check-box {
  @apply w-6 h-6;
}

.checkbox.small .check-box {
  @apply w-4 h-4;
}

.checkbox.small .check-box,
.checkbox .check-box {
  @apply border border-grey-shd4 block rounded-sm flex items-center justify-center transition duration-100 ease-in-out hover:border-grey-shd1 relative;
  z-index: 1;
}

.checkbox .check-box .check-mark {
  @apply hidden;
}

.checkbox .checkbox-input[disabled] + .check-box {
  @apply border-grey-shd6 hover:border-grey-shd6;
}

.checkbox .checkbox-input[disabled][checked] + .check-box {
  @apply bg-primary-shd7 hover:bg-primary-shd7;
}

/* Checkboxes */
