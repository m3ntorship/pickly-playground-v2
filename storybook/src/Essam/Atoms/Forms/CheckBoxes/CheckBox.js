import React from "react";
import PropTypes from "prop-types";
import className from "classnames";
import "./CheckBox.css";

export const CheckBox = ({ id, checkBoxSize, ...props }) => {
  const inputId = id.trim().split(" ").join("");

  const CkeckBoxClasses = className("checkbox", {
    small: checkBoxSize === "small",
  });
  return (
    <div className={CkeckBoxClasses}>
      <input
        type="checkbox"
        id={inputId}
        className="checkbox-input"
        {...props}
      />
      <label htmlFor={inputId} className="check-box">
        <div className="check-mark">
          <svg
            width="16"
            height="16"
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
              fill="white"
            />
          </svg>
        </div>
      </label>
    </div>
  );
};

CheckBox.propTypes = {
  id: PropTypes.string,
  checkBoxSize: PropTypes.oneOf(["", "small"]),
};
CheckBox.defaultProps = {
  id: "",
  checkBoxSize: "",
};
