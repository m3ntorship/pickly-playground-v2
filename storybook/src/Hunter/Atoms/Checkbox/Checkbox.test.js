import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CheckBox from "./CheckBox";

describe("checkbox component", () => {
  it("should render (default-unchecked) checkedbox when we apply nothing", () => {
    const tree = renderer.create(<CheckBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (disabled) checkedbox when we apply disabled", () => {
    const tree = renderer.create(<CheckBox disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (small) checkedbox when we apply small", () => {
    const tree = renderer.create(<CheckBox size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // it ("should render (checked) checkedbox when we apply cjecked", ()=>{\
  //     const tree = renderer.create(<CheckBox  />).toJSON();
  //     expect(tree).toMatchSnapshot();
  // });
});

test("checkbox checked or not", () => {
  const { getByTestId } = render(<CheckBox />);
  expect(getByTestId("checkbox")).not.toBeChecked();
});
test("checkbox checked after clicked", () => {
  const { getByTestId } = render(<CheckBox />);
  const checkBox = getByTestId("checkbox");
  userEvent.click(checkBox);
  expect(checkBox).toBeChecked();
});
test("checkbox checked or not", () => {
  const { getByTestId } = render(<CheckBox disabled />);
  expect(getByTestId("checkbox")).toBeDisabled();
});
