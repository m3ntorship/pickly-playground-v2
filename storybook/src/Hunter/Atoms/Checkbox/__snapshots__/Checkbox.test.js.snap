// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox component should render (default-unchecked) checkedbox when we apply nothing 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    data-testid="checkbox"
    type="checkbox"
  />
  <div
    className="cursor-pointer border border-grey-shd4 flex items-center justify-center hover:border-grey-shd1 transition-all duration-500 ease-out rounded-sm 
                    
                    bg-white    
                    w-6 h-6     
                    "
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="w-4 h-4 "
      viewBox="0 0 16 16"
      width="w-4 h-4 "
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`checkbox component should render (disabled) checkedbox when we apply disabled 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    data-testid="checkbox"
    type="checkbox"
  />
  <div
    className="cursor-pointer border border-grey-shd4 flex items-center justify-center hover:border-grey-shd1 transition-all duration-500 ease-out rounded-sm 
                    pointer-events-none opacity-25
                    bg-white    
                    w-6 h-6     
                    "
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="w-4 h-4 "
      viewBox="0 0 16 16"
      width="w-4 h-4 "
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`checkbox component should render (small) checkedbox when we apply small 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    data-testid="checkbox"
    type="checkbox"
  />
  <div
    className="cursor-pointer border border-grey-shd4 flex items-center justify-center hover:border-grey-shd1 transition-all duration-500 ease-out rounded-sm 
                    
                    bg-white    
                    w-4 h-4     
                    "
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="w-3 h-3 "
      viewBox="0 0 16 16"
      width="w-3 h-3 "
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;
